document/CoreFormat.mdk

```diff
@@ -826,12 +791,111 @@ In this case, it is helpful to the author of the logging component to know
+
+### `commandLine` property { #invocation-commandLine }
+
+An `invocation` object may contain a property named `commandLine` whose value is a string
+containing the runtime parameters with which the tool was invoked.
+For command line tools, this string may consist of the completely specified command line used to invoke the tool.
+
+~ IsoNote
+The information in the `commandLine` property makes it possible to precisely repeat a run of an analysis tool,
+and to verify that the results reported in the log file were generated by an appropriate invocation of the tool.
+~
+
+If the information in `commandLine` contains information which should not be disclosed,
+such as passwords, tokens, database connection strings, or in some circumstances even the fully qualified path to
+the tool executable, that information should be redacted or omitted.
+Redacted information should be replaced with the token `[REMOVED]`.
+
+~ IsoExample
+Suppose a tool is invoked with the command line
+
+```
+    C:\Users\johnsmith\Tools\DbScanner\DbScanner.exe
+        /ConnectionString "Server=CorpServer;Database=Accounting;User Id=Admin;Password=S3cr#t" /input *.sql
+```
+
+Then the value of the `commandLine` property might contain the redacted command line
+
+```
+    [REMOVED]\DbScanner.exe /connectionString=[REMOVED] /input=*.sql 
+```
+~
+
+~ IsoNote
+Redacting sensitive information from `commandLine` makes it more difficult to precisely reproduce
+an analysis run. The value of `commandLine` would have to be combined with information from another
+source to allow the run to be repeated.
+~
+
+### `machine` property { #invocation-machine }
+
+An `invocation` object may contain a property named `machine` whose value is a string containing
+the name of the machine on which the tool was run.
+
+### `account` property { #invocation-account }
+
+An `invocation` object may contain a property named `account` whose value is a string containing
+the name of the account under which the tool was run.
+
+### `processId` property { #invocation-processId }
+
+An `invocation` object may contain a property named `processId` whose value is an integer containing
+the id of the process in which the tool was run.
+
+### `fileName` property { #invocation-fileName }
+
+An `invocation` object may contain a property named `fileName` whose value is a string containing
+the fully qualified path name of the tool's executable file.
+
+~ IsoNumberedNote { @IsoNumberedNote=1 }
+This property is defined in the `invocation` object rather than in the `tool` object (&sect;[#tool]) because the
+identical tool might be invoked from different paths on different machines.
+~
+
+~ IsoNumberedNote
+This property might duplicate information in the `commandLine` property (&sect;[#invocation-commandLine]).
+Storing the file name in a separate property eliminates the need to parse the command line to isolate
+the file name.
```

Comment:
This isn't really the rationale I'd emphasize. Rather, I'd note that the commandLine is intended to express the literal commandLine used for execution (to assist in troubleshooting). Because that information doesn't necessarily include the path to the exe that ran and because that information could be vital for debugging some scenarios, we create a place for it here.